compiler/src/lntoamm/Statement.ts

```diff
@@ -88,64 +91,15 @@ class Statement {
     return true
   }
 
-  static isWithOperatorsPure(withOperatorsAst: any, scope: Scope) { // TODO: Migrate off ANTLR
-    for (const operatorOrAssignable of withOperatorsAst.operatororassignable()) {
-      if (operatorOrAssignable.operators() != null) {
-        const operator = operatorOrAssignable.operators()
-        const op = scope.deepGet(operator.getText()) as Array<Operator>
-        if (!op || !(op instanceof Array && op[0] instanceof Operator)) {
-          throw new Error("Operator " + operator.getText() + " is not defined")
-        }
-        // TODO: Similar to the above, need to figure out logic to determine which particular function
-        // will be the one called. For now, just assume the first one and fix this later.
-        if (!op[0].potentialFunctions[0].isPure()) return false
-      }
-      if (operatorOrAssignable.basicassignables() != null) {
-        if (!Statement.isBasicAssignablePure(operatorOrAssignable.basicassignables(), scope)) {
-          return false
-        }
-      }
-    }
-
-    return true
-  }
-
-  static isBasicAssignablePure(basicAssignable: any, scope: Scope) { // TODO: Migrate off ANTLR
-    if (basicAssignable.functions() != null) {
-      // Defining a function in itself is a pure situation
-      return true
-    }
-    if (basicAssignable.calls() != null) {
-      return Statement.isCallPure(basicAssignable.calls(), scope)
-    }
-    if (basicAssignable.varn() != null) {
-      // This would be a read-only operation to pull a value into local scope
-      return true
-    }
-    if (basicAssignable.constants() != null) {
-      // This is an explicit constant that cannot impact any outer scope
-      return true
-    }
-    if (basicAssignable.groups() != null) {
-      // This is a "group" (parens surrounding one or more operators and operands)
-      return Statement.isWithOperatorsPure(basicAssignable.groups().withoperators(), scope)
-    }
-    // Shouldn't be reached
-    return false
-  }
-
   static isAssignablePure(assignableAst: any, scope: Scope) { // TODO: Migrate off ANTLR
-    if (assignableAst.basicassignables() != null) {
-      return Statement.isBasicAssignablePure(assignableAst.basicassignables(), scope)
-    }
-    if (assignableAst.withoperators() != null) {
-      return Statement.isWithOperatorsPure(assignableAst.withoperators(), scope)
-    }
-    // This should never be reached
-    throw new Error("Impossible assignment situation")
+    // TODO: Redo this
```

Comment:
nit / not blocking this PR: what is "this" and what is a potential Redo? I actually have hit this error a few times

Reply:
I deleted everything and just marked all assignables pure all the time. So it needs to be completely redone and I thought it was self-evident that `return true` is not something you'd ever actually want from a function call.

We aren't yet using function/statement purity in the compiler, so it's not a huge deal, but if/when we want the compiler to automatically downgrade a `map` to a `mapLin` because you're mutating outer state, this will need to be working and correct.

I can update the TODO comment with all of this if you want?

Reply:
Since it's not blocking, though, I can do it in a followup PR.